pkg/appliance/backup/backup.go

```diff
@@ -86,6 +88,70 @@ func (b *Backup) Download(ctx context.Context, applianceID, backupID, destinatio
 	return out, nil
 }
 
+// Download a completed appliance backup with the given ID of an Appliance
+func (b *Backup) Download(ctx context.Context, applianceID, backupID, destination string) (*os.File, error) {
+	out, err := os.Create(destination)
+	if err != nil {
+		return nil, err
+	}
+	defer out.Close()
+
+	client := b.HTTPClient
+	cfg := b.APIClient.GetConfig()
+	url, err := cfg.ServerURLWithContext(ctx, "ApplianceBackupApiService.AppliancesIdBackupBackupIdGet")
+	if err != nil {
+		return nil, err
+	}
+	url = fmt.Sprintf("%s/appliances/%s/backup/%s", url, applianceID, backupID)
+	ctx = context.WithValue(ctx, api.ContextAcceptValue, fmt.Sprintf("application/vnd.appgate.peer-v%d+gpg", b.Version))
+
+	headReq, err := http.NewRequestWithContext(ctx, http.MethodHead, url, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	head, err := client.Do(headReq)
+	if err != nil {
+		return nil, err
+	}
+
+	size := head.ContentLength
+	if err := retryDownload(ctx, client, out, url, size); err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func retryDownload(ctx context.Context, client *http.Client, f *os.File, url string, size int64) error {
+	start := int64(0)
+	buf := make([]byte, size)
```

Comment:
Does that mean you are allocating a buffer the size of the whole file?
Wouldn't it be better to read in chunks and write directly to the file without accumulating everything in memory.
While still keeping track of `start` for purposes of the range header.

Reply:
I agree I'll see what I can do here.